mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoImplementationGenerator.java

```diff
@@ -15,33 +15,67 @@
  */
 package com.datastax.oss.driver.internal.mapper.processor.dao;
 
+import com.datastax.oss.driver.api.mapper.annotations.EntityInjector;
+import com.datastax.oss.driver.api.mapper.entity.EntityHelper;
 import com.datastax.oss.driver.internal.core.util.concurrent.BlockingOperation;
 import com.datastax.oss.driver.internal.core.util.concurrent.CompletableFutures;
 import com.datastax.oss.driver.internal.mapper.MapperContext;
 import com.datastax.oss.driver.internal.mapper.processor.GeneratedNames;
+import com.datastax.oss.driver.internal.mapper.processor.PartialClassGenerator;
 import com.datastax.oss.driver.internal.mapper.processor.ProcessorContext;
 import com.datastax.oss.driver.internal.mapper.processor.SingleFileCodeGenerator;
+import com.datastax.oss.driver.internal.mapper.processor.SkipGenerationException;
+import com.datastax.oss.driver.internal.mapper.processor.util.NameIndex;
 import com.squareup.javapoet.ClassName;
 import com.squareup.javapoet.FieldSpec;
 import com.squareup.javapoet.JavaFile;
 import com.squareup.javapoet.MethodSpec;
 import com.squareup.javapoet.ParameterizedTypeName;
 import com.squareup.javapoet.TypeSpec;
+import java.beans.Introspector;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
 import javax.lang.model.element.Modifier;
 import javax.lang.model.element.TypeElement;
 
 public class DaoImplementationGenerator extends SingleFileCodeGenerator {
 
   private final TypeElement interfaceElement;
   private final ClassName implementationName;
+  private final NameIndex nameIndex = new NameIndex();
+  private final Map<ClassName, String> entityHelperFields = new HashMap<>();
 
   public DaoImplementationGenerator(TypeElement interfaceElement, ProcessorContext context) {
     super(context);
     this.interfaceElement = interfaceElement;
     implementationName = GeneratedNames.daoImplementation(interfaceElement);
   }
 
+  /**
+   * Requests the generation of a field holding the {@link EntityHelper} that was generated for the
+   * given entity class, along with the initialization code in the constructor.
+   *
+   * <p>If this is called multiple times, only a single field will be created.
+   *
+   * @return the name of the field.
+   */
+  String addEntityHelperField(TypeElement entityClass) {
+    ClassName helperClass = GeneratedNames.entityHelper(entityClass);
+    return entityHelperFields.computeIfAbsent(
+        helperClass,
+        k -> {
+          String baseName =
+              Introspector.decapitalize(entityClass.getSimpleName().toString()) + "Helper";
```

Comment:
The name will be generated in a camel-case pattern?

Reply:
I see a decapatlize here am I missing something?.. why would this be camel case?

Reply:
Yes, it lower-cases the first letter.
```
Product => productHelper
SomeClass => someClassHelper
```